x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/export/get_export_by_object_ids.test.ts

```diff
@@ -52,542 +42,285 @@ const connectors = [
-            outcome: undefined,
-            alias_target_id: undefined,
-            alias_purpose: undefined,
-            timestamp_override: undefined,
-            timestamp_override_fallback_disabled: undefined,
-            namespace: undefined,
-            data_view_id: undefined,
-            alert_suppression: undefined,
-            investigation_fields: undefined,
+    expect(JSON.parse(exports.rulesNdjson)).toMatchObject({
+      actions: [
+        {
+          group: 'default',
+          id: 'preconfigured-connector',
+          params: {
+            message: 'Rule {{context.rule.name}} generated {{state.signals_count}} alerts',
           },
-        ],
-      };
-      expect(exports).toEqual(expected);
+          action_type_id: '.email',
+          frequency: { summary: true, throttle: null, notifyWhen: 'onActiveAlert' },
+        },
+      ],
     });
+    expect(JSON.parse(exports.exportDetails)).toMatchObject({
+      exported_count: 1,
+      exported_rules_count: 1,
+      exported_action_connector_count: 0,
+      missing_action_connection_count: 0,
+      missing_action_connections: [],
+    });
+    expect(exports.actionConnectors).toBe('');
+  });
 
-    test('it does not transform the rule if the rule is an immutable rule and designates it as a missing rule', async () => {
-      const rulesClient = rulesClientMock.create();
-      const result = getRuleMock(getQueryRuleParams());
-      result.params.immutable = true;
-
-      const findResult: FindHit = {
-        page: 1,
-        perPage: 1,
-        total: 0,
-        data: [result],
-      };
-
-      rulesClient.get.mockResolvedValue(result);
-      rulesClient.find.mockResolvedValue(findResult);
+  test('it processes large exports in chunks to avoid "too_many_clauses" error', async () => {
+    const EXPECTED_CHUNK_SIZE = 1024;
+    // Let's have 3 chunks, two chunks by 1024 rules and the third chunk containing just one rule
+    const RULES_COUNT = 2 * EXPECTED_CHUNK_SIZE + 1;
+    const rules = new Array(RULES_COUNT)
+      .fill(0)
+      .map((_, i) => getRuleMock(getQueryRuleParams({ ruleId: `rule-${i}` })));
 
-      const objects = [{ rule_id: 'rule-1' }];
-      const exports = await getRulesFromObjects(
-        rulesClient,
-        clients.savedObjectsClient,
-        objects,
-        logger
-      );
-      const expected: RulesErrors = {
-        exportedCount: 0,
-        missingRules: [{ rule_id: 'rule-1' }],
-        rules: [],
-      };
-      expect(exports).toEqual(expected);
+    const rulesClient = rulesClientMock.create();
+    const chunk1 = getFindResultWithMultiHits({
+      data: rules.slice(0, EXPECTED_CHUNK_SIZE),
+    });
+    const chunk2 = getFindResultWithMultiHits({
+      data: rules.slice(EXPECTED_CHUNK_SIZE, 2 * EXPECTED_CHUNK_SIZE),
+    });
+    const chunk3 = getFindResultWithMultiHits({
+      data: rules.slice(2 * EXPECTED_CHUNK_SIZE),
     });
 
-    test('it exports missing rules', async () => {
```

Comment:
I think a test for this case is missing in the refactored version. I suggest to add it.